### DISCUSSION TITLE
How can I get the e-mail address of the committer?

### DISCUSSION THREAD

**Discussion by ohsayan** [community/community#25448] [Actions]
<p>So I have a workflow in which a push (by the merge of a PR or a simple push) would trigger a bot to push to another repo with the user’s name and details, that triggered the action. I managed to get the name by <code>{{ github.event.inputs.name }}</code>. But what about the user’s e-mail?</p>

**Comment by dawidd6** [comment, marked as answer]
<p>For the <code>push</code> event, there is: <code>${{github.event.pusher.email}}</code>.</p>
<p>See: <a href="https://developer.github.com/webhooks/event-payloads/#push" rel="noopener nofollow ugc">https://developer.github.com/webhooks/event-payloads/#push</a></p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/ohsayan">@ohsayan</a>,</p>
<p>If the workflow is triggered by push event, you can use the following properties to get the email and name of the pusher from the <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions#github-context" rel="noopener nofollow ugc">github context</a> of the workflow run.</p>
<blockquote>
<p><strong>github.event.pusher.email</strong><br>
<strong>github.event.pusher.name</strong></p>
</blockquote>
<p>For example:</p>
<pre><code class="lang-yaml">on: push

jobs:
  job1:
    runs-on: ubuntu-latest
    steps:
      - name: View the information of the pusher
        run: |
          echo "The name of the pusher: ${{ github.event.pusher.name }}"
          echo "The email of the pusher: ${{ github.event.pusher.email }}"

      - name: View the github context
        run: echo "$GITHUB_CONTEXT"
        env:
          GITHUB_CONTEXT: ${{ toJson(github) }}
</code></pre>

**Comment by Simran-B** [comment]
<p><a class="mention" href="/dawidd6">@dawidd6</a>’s link points to the legacy documentation, the new link is:<br>
<a href="https://docs.github.com/en/free-pro-team@latest/developers/webhooks-and-events/webhook-events-and-payloads#push" rel="noopener nofollow ugc">https://docs.github.com/en/free-pro-team@latest/developers/webhooks-and-events/webhook-events-and-payloads#push</a></p>
